tests/migration/test_config_map_data_handler.py

```diff
@@ -0,0 +1,744 @@
+#
+# MIT License
+#
+# (C) Copyright 2023 Hewlett Packard Enterprise Development LP
+#
+# Permission is hereby granted, free of charge, to any person obtaining a
+# copy of this software and associated documentation files (the "Software"),
+# to deal in the Software without restriction, including without limitation
+# the rights to use, copy, modify, merge, publish, distribute, sublicense,
+# and/or sell copies of the Software, and to permit persons to whom the
+# Software is furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included
+# in all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+# THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+# OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+# ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+# OTHER DEALINGS IN THE SOFTWARE.
+#
+
+"""
+Tests for validating ConfigMapDataHandler
+"""
+
+import unittest
+from unittest.mock import patch, call, Mock
+from typing import Dict, List
+
+from kubernetes.config import ConfigException
+from kubernetes import client
+from kubernetes.client.api_client import ApiClient
+
+from cray_product_catalog.migration.main import main
+from cray_product_catalog.migration.config_map_data_handler import ConfigMapDataHandler
+from cray_product_catalog.migration.kube_apis import KubernetesApi
+from cray_product_catalog.constants import (
+    PRODUCT_CATALOG_CONFIG_MAP_NAME, PRODUCT_CATALOG_CONFIG_MAP_NAMESPACE,
+    PRODUCT_CATALOG_CONFIG_MAP_LABEL
+)
+from cray_product_catalog.util.catalog_data_helper import format_product_cm_name
+from tests.mock_update_catalog import (
+    ApiInstance, ApiException
+)
+from cray_product_catalog.migration import CONFIG_MAP_TEMP
+
+YAML_DATA = """

```

Comment:
shall we move these to separate file `migration_mock.py`? or we will keep it here?
@krabhi-hpe @lathanm

Reply:
moved